Tests/Test_FaultThread_Exceptions.c

```diff
@@ -0,0 +1,425 @@
+        // Both with and without callback functions
+        printf("\n\r=========== Testing exception priorities 1 and 2 ===========");
+    
+        // Test level 1 & 2 exceptions without callbacks
+        createExceptionTask(NULL); 
+        checkOSError(err); 
+        createFaultState();
+        checkOSError(err); 
+        OSSemPend(&Ready_Sema4, 0, OS_OPT_PEND_BLOCKING, &ts, &err); // Wait for task to finish
+        checkOSError(err); 
+        OSTimeDlyHMSM(0, 0, 1, 0, OS_OPT_TIME_HMSM_STRICT, &err); // extra time for the task finished
+        checkOSError(err); 
+        OSTaskDel(&ExceptionTaskTCB, &err);
+        checkOSError(err); 
+        OSTaskDel(&FaultState_TCB, &err);
+        checkOSError(err); 
+        
+        // Test level 1 & 2 exceptions with callbacks
+        createExceptionTask(exceptionCallback); 
+        checkOSError(err); 
+        createFaultState();
+        checkOSError(err); 
+        OSSemPend(&Ready_Sema4, 0, OS_OPT_PEND_BLOCKING, &ts, &err);
+        OSTimeDlyHMSM(0, 0, 1, 0, OS_OPT_TIME_HMSM_STRICT, &err);
+        checkOSError(err); 
+        OSTaskDel(&ExceptionTaskTCB, &err);
+        checkOSError(err); 
+        OSTaskDel(&FaultState_TCB, &err);
+        checkOSError(err); 
+
+        // Test the assertOSError function using the OSErrorTask
+        printf("\n\r=========== Testing OS assert ===========");
+
+        createOSErrorTask();
+        checkOSError(err); 
+        createFaultState();
+        checkOSError(err); 
+        OSSemPend(&Ready_Sema4, 0, OS_OPT_PEND_BLOCKING, &ts, &err);
+        checkOSError(err);
+        OSTimeDlyHMSM(0, 0, 0, 500, OS_OPT_TIME_HMSM_STRICT, &err);
+        checkOSError(err); 
+        OSTaskDel(&OSErrorTaskTCB, &err);
+        checkOSError(err); 
+        OSTaskDel(&FaultState_TCB, &err);
+
+        // Test individual tasks error assertions
+        printf("\n\r=========== Testing ReadTritium ===========");
+
+        CANDATA_t canMessage = {0};
+        canMessage.ID = CARDATA;
+        canMessage.idx = 0;
+        *(uint16_t*)(&canMessage.data[0]) = READY_MSG;
+
+        for (int i = 0; i < NUM_TRITIUM_ERRORS + 2; i++) {
+            createFaultState();
+            createReadTritium();
+            // Send a message to the motor to tell it to start sending stuff
+            printf("\n\rShould be testing Tritium error %d", i);
+            CANbus_Send(canMessage, CAN_BLOCKING, MOTORCAN);
+            OSTimeDlyHMSM(0, 0, 0, 100, OS_OPT_TIME_HMSM_STRICT, &err); // Wait for ReadTritium to finish
+            OSTaskDel(&ReadTritium_TCB, &err);
+            OSTaskDel(&FaultState_TCB, &err);
+
+        }
+
+        printf("\n\r=========== Testing ReadCarCAN ===========");
+        createFaultState();
+        checkOSError(err);
+        createReadCarCAN();
+        checkOSError(err);
+        CANbus_Send(canMessage, CAN_BLOCKING, CARCAN);
+        uint16_t canMsgData = 0;
+
+        do {
```

Comment:
This part of the test seems to get stuck in an infinite loop of faults and I can't quite figure out why. What is it supposed to be doing?

Reply:
Right now that's all it does. I've been struggling to debug the tests, so right now every test starting with Read Tritium onwards is nonfunctional. I think the infinite loop of "read BPS error" is the callback function of the ReadCarCAN Watchdog timer. I'm sorry I wasted your time trying to figure out what was going on

Reply:
Everything up to it seems to work. I can try testing ReadCarCAN and onward next week